### PR TITLE
[Feat] Deprecate ForcedClusterUpgrade

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Why are these changes needed?

Deprecate `ForcedClusterUpgrade`. Use `RayClusterSpec.Suspend` instead.

## Related issue number

N/A

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kevin85421** [comment]
Let's chat this PR in this week's sync.

File: ray-operator/controllers/ray/raycluster_controller.go [hunk 1]
```diff
@@ -697,42 +696,6 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 		}
 	}
 
-	if ForcedClusterUpgrade {
-		if len(headPods.Items) == 1 {
-			// head node amount is exactly 1, but we need to check if it has been changed
-			res := utils.PodNotMatchingTemplate(headPods.Items[0], instance.Spec.HeadGroupSpec.Template)
```
Comment (kevin85421) [inline ray-operator/controllers/ray/raycluster_controller.go:703]: You can check whether there is any other place referring to this function. If not, we can remove it.

Comment (MortalHappiness) [reply ray-operator/controllers/ray/raycluster_controller.go:703]: `utils.PodNotMatchingTemplate` is used in https://github.com/ray-project/kuberay/blob/a91fc5a217b092b61126945745db777466186029/ray-operator/controllers/ray/utils/util_test.go#L301
so I do not remove it.

Comment (kevin85421) [reply ray-operator/controllers/ray/raycluster_controller.go:703]: The test seems to be entirely for `PodNotMatchingTemplate` (https://github.com/ray-project/kuberay/pull/231). Maybe we could also remove it?

Comment (MortalHappiness) [reply ray-operator/controllers/ray/raycluster_controller.go:703]: Done

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**